tests/unit/test_actions.py

```diff
@@ -75,3 +74,28 @@ def test_targeted_error_object_with_dataset_sensitivity():
     error = result.errors[0].to_representation()
     assert "row" not in error
     assert error["value"] == {"TEST": 1, "MISSING": "Not in dataset"}
+
+
+def test_empty_sequential_not_raise_errors():
+    dummy_rule = {
+        "core_id": "MockRule",
+        "actions": [
+            {
+                "name": "generate_dataset_error_objects",
+                "params": {
+                    "message": "TEST greater than 2",
+                },
+            }
+        ],
+        "output_variables": ["TV"],
+    }
+    df = pd.DataFrame.from_dict(
+        {"TVSEQ": [2, 4, 6, None, "", 8], "TV": [1, 3, 5, 7, 9, "8"]}
+    )
+    variable = DatasetVariable(df)
+    action = COREActions([], variable, "TV", dummy_rule)
+    targets = set(dummy_rule["output_variables"])
+    result = action.generate_targeted_error_object(targets, df, "TVSEQ greater than 2")
+    assert result.errors[3]._sequence is None
```

Comment:
Instead of checking the protected attribute, check the whole result. Create the expected result and compare it with the actual result.